drivers/entropy/entropy_cc3xx.c

```diff
@@ -29,16 +29,19 @@
 static int entropy_cc3xx_rng_get_entropy(const struct device *dev,
 					 uint8_t *buffer, uint16_t length)
 {
-	int res = -EINVAL;
-
 	__ASSERT_NO_MSG(dev != NULL);
 	__ASSERT_NO_MSG(buffer != NULL);
+	int err = EINVAL;
 
 #if defined(CONFIG_BUILD_WITH_TFM)
-	res = psa_generate_random(buffer, length);
-	if (res != PSA_SUCCESS) {
-		return -EINVAL;
+	psa_status_t status;
+
+	status = psa_generate_random(buffer, length);
+	if (status != PSA_SUCCESS) {
+		err = 0;
```

Comment:
Shouldn't `0` be returned when `status == PSA_SUCCESS`?

IMHO this `err` variable does not really help, so I'd suggest to get rid of it and simply return `-EINVAL` here and `0` below. And similarly for the non-TFM path - return an error when something goes wrong and `0` at the end.

Reply:
Thanks. Fixed.